File: clients/ios/TmuxRemote/Patterns/PatternMatch.swift
import Foundation

struct ResolvedAction: Identifiable, Equatable {
    let id = UUID()
    let label: String
    let keys: String
}

struct PatternMatch: Identifiable, Equatable {
    let id: String              // instance_id
    let patternId: String
    let patternType: PatternType
    let prompt: String?
    let actions: [ResolvedAction]
    let revision: Int

    static func == (lhs: PatternMatch, rhs: PatternMatch) -> Bool {
        lhs.id == rhs.id && lhs.revision == rhs.revision
    }
}

File: clients/android/app/src/main/java/com/nabto/tmuxremote/patterns/PatternMatch.kt
package com.nabto.tmuxremote.patterns

import java.util.UUID

enum class PatternType(val value: String) {
    YES_NO("yes_no"),
    NUMBERED_MENU("numbered_menu"),
    ACCEPT_REJECT("accept_reject");

    companion object {
        fun fromString(s: String): PatternType? = entries.find { it.value == s }
    }
}

data class ResolvedAction(
    val id: String = UUID.randomUUID().toString(),
    val label: String,
    val keys: String
)

data class PatternMatch(
    val id: String,           // instance_id
    val patternId: String,
    val patternType: PatternType,
    val prompt: String?,
    val actions: List<ResolvedAction>,
    val revision: Int
)
